test/Microsoft.AspNetCore.Server.KestrelTests/BadHttpRequestTests.cs

```diff
@@ -52,6 +52,12 @@ public class BadHttpRequestTests
         [InlineData("} / HTTP/1.0\r\n")]
         [InlineData("get@ / HTTP/1.0\r\n")]
         [InlineData("post= / HTTP/1.0\r\n")]
+        // malformed request target
+        [InlineData("GET http:// HTTP/1.1\r\n")]
+        [InlineData("GET https:// HTTP/1.1\r\n")]
+        [InlineData("GET http:/// HTTP/1.1\r\n")]
+        [InlineData("GET http://// HTTP/1.1\r\n")]
```

Comment:
Add:

- Not starting with `http://` or `https://`
- `http`
- `https`
- `http:`
- `https:/`
- `http:/`
- `https:/`
- `http:// ` (note the extra space)
- `https:// `
- `http:// /`
- `https:// /`